src/promptflow-tools/tests/conftest.py

```diff
@@ -44,15 +44,26 @@ def serp_connection():
     return ConnectionManager().get("serp_connection")
 
 
+@pytest.fixture
+def gpt2_custom_connection():
+    return ConnectionManager().get("custom_connection")
+
+
 @pytest.fixture(autouse=True)
 def skip_if_no_key(request, mocker):
     mocker.patch.dict(os.environ, {"PROMPTFLOW_CONNECTIONS": CONNECTION_FILE})
     if request.node.get_closest_marker('skip_if_no_key'):
         conn_name = request.node.get_closest_marker('skip_if_no_key').args[0]
         connection = request.getfixturevalue(conn_name)
         # if dummy placeholder key, skip.
-        if "-api-key" in connection.api_key:
-            pytest.skip('skipped because no key')
+        try:
+            if "-api-key" in connection.api_key:
```

Comment:
Maybe could add conditions like `isinstance(connection, OpenAIConnection)` to replace `try-except`